### PR TITLE
Fixup and optimize code

### PR DESCRIPTION
**Pull request by queengooborg**
This PR performs numerous optimizations and fixes to our code, fixing issues within our linter scripts and parallelizing some functions.

This PR was created as an experiment of GitHub Copilot to assess its capabilities and usefulness. The majority of changes were generated by GitHub Copilot.

Depends on #21552.

### CODE REVIEW

File: scripts/fix/links.ts [hunk 1]
```diff
@@ -11,8 +11,8 @@ import { processData } from '../../test/linter/test-links.js';
  * @param {string} filename The name of the file to fix
  */
 const fixLinks = (filename: string): void => {
-  const errors = processData(filename);
   const original = fs.readFileSync(filename, 'utf-8').trim();
+  const errors = processData(original);
```
Comment (teoli2003) [inline scripts/fix/links.ts:15]: If I understand well this change, this function never worked?

Comment (queengooborg) [reply scripts/fix/links.ts:15]: Yep, pretty much...

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Globally this looks good. I have 2 questions that I expected to be positively answered and this can be merged.

File: scripts/release/build.ts [hunk 2]
```diff
@@ -209,11 +208,13 @@ const main = async () => {
...
-  await writeWrapper();
-  await writeTypeScript();
-  await copyFiles();
+  await Promise.all([
+    writeManifest(),
+    writeData(),
+    writeWrapper(),
+    writeTypeScript(),
+    copyFiles(),
+  ]);
```
[the remark this answers is not in the document]
Comment (queengooborg) [reply scripts/release/build.ts:217]: Yep, all of the functions modify completely different files, so they can all be done in parallel!

[the remark this answers is not in the document]
Comment (teoli2003) [reply scripts/release/build.ts:217]: That was my understanding!

**Review by teoli2003 — APPROVED** [review]

**Pull request merged**